extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -467,7 +465,30 @@ public function getCuratedContentSection() {
 			throw new NotFoundApiException( 'No members' );
 		}
 
-		$this->response->setVal( 'items', $data[ 'items' ] );
+		$this->response->setVal( 'items', $data['items'] );
+	}
+
+	public function getMainPageDetailsAndAdsContext() {
+		$mainPageTitle = Title::newMainPage();
+		$mainPageArticleID = $mainPageTitle->getArticleID();
+		$article = Article::newFromID( $mainPageArticleID );
+
+		$articleDetails = $this->getArticleDetails( $article );
+
+		if ( empty( $articleDetails ) ) {
+			throw new WikiaException( 'Article details empty', 500 );
```

Comment:
We don't throw those exceptions in `getArticle` so why here? User can still use the page without them, right? Maybe we should just log an error?